File: CryptoBridge-node/src/infrastructure/external-apis/MercadoBitcoinClient.js
const axios = require('axios');
const Ticker = require('../../domain/entities/Ticker');

class MercadoBitcoinClient {
    constructor() {
        this.BaseUrl = 'https://api.mercadobitcoin.net/api/v4';
        this.timeout = 10000; // 10 seconds
    }

    async getSymbols() {
        try {
            const response = await axios.get(`${this.BaseUrl}/symbols`, {
                timeout: this.timeout
            });

            if (response.status !== 200) {
                return [];
            }

            return response.data.symbol || [];
        } catch (error) {
            console.error('Erro ao buscar símbolos no Mercado Bitcoin:', error.message);
            return [];
        }
    }

    async getTicker(symbol) {
        try {
            const response = await axios.get(`${this.BaseUrl}/tickers?symbols=${symbol}`, {
                timeout: this.timeout
            });

            if (response.status !== 200) {
                return null;
            }

            const data = response.data;

            if (!data || data.length === 0) {
                return null;
            }

            const tickerData = data[0];

            return new Ticker(
                symbol,
                parseFloat(tickerData.buy || 0),
                parseFloat(tickerData.sell || 0),
                parseFloat(tickerData.vol || 0),
                parseFloat(tickerData.low || 0),
                parseFloat(tickerData.high || 0),
                parseFloat(tickerData.last || 0)
            );
        } catch (error) {
            console.error(`Erro ao buscar ticker ${symbol} no Mercado Bitcoin:`, error.message);
            return null;
        }
    }
}

module.exports = MercadoBitcoinClient;

File: CryptoBridge-python/src/infrastructure/external_apis/mercado_bitcoin_client.py
import asyncio
from typing import List, Optional, Dict, Any
import httpx
from ...domain.entities.ticker import Ticker


class MercadoBitcoinClient:
    BaseUrl = "https://api.mercadobitcoin.net/api/v4"

    def __init__(self):
        self.timeout = httpx.Timeout(10.0, connect=5.0)

    async def getSymbols(self) -> List[str]:
        try:
            async with httpx.AsyncClient(timeout=self.timeout) as client:
                response = await client.get(f"{self.BaseUrl}/symbols")

                if response.status_code != 200:
                    return []

                data = response.json()
                return data.get("symbol", [])

        except Exception as e:
            print(f"Erro ao buscar símbolos no Mercado Bitcoin: {e}")
            return []

    async def getTicker(self, symbol: str) -> Optional[Ticker]:
        try:
            async with httpx.AsyncClient(timeout=self.timeout) as client:
                response = await client.get(f"{self.BaseUrl}/tickers?symbols={symbol}")

                if response.status_code != 200:
                    return None

                data = response.json()

                if not data or len(data) == 0:
                    return None

                ticker_data = data[0]

                return Ticker(
                    symbol=symbol,
                    buy=float(ticker_data.get("buy", 0)),
                    sell=float(ticker_data.get("sell", 0)),
                    vol=float(ticker_data.get("vol", 0)),
                    low=float(ticker_data.get("low", 0)),
                    high=float(ticker_data.get("high", 0)),
                    last=float(ticker_data.get("last", 0))
                )

        except Exception as e:
            print(f"Erro ao buscar ticker {symbol} no Mercado Bitcoin: {e}")
            return None